docs/DeveloperGuide.md

```diff
@@ -1,38 +1,325 @@
 # Developer Guide
 
+## Table of Contents
+
++ [Acknowledgements](#acknowledgements)
++ [Design](#design)
+    * [Architecture](#architecture)
+    * [Parser Component](#parser-component)
+    * [UI Component](#ui-component)
+    * [Storage Component](#storage-component)
+    * [ExpenseManager Component](#expensemanager-component)
++ [Implementation](#implementation)
++ [Product Scope](#product-scope)
+    * [Target User Profile](#target-user-profile)
+    * [Value Proposition](#value-proposition)
++ [User Stories](#user-stories)
++ [Non-Functional Requirements](#non-functional-requirements)
++ [Glossary](#glossary)
++ [Instructions for Manual Testing](#instructions-for-manual-testing)
+
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
+
+## Design
+
+### Architecture
```

Comment:
Adding an architecture diagram so developers have an overview of how the application works would be helpful!